elrond-wasm/src/esdt.rs

```diff
@@ -29,12 +33,112 @@ impl<BigUint: BigUintApi> ESDTSystemSmartContractProxy<BigUint> {
 	}
 
 	/// Produces a contract call to the ESDT system SC,
-	/// which causes it to issue a new ESDT token.
-	pub fn issue(
+	/// which causes it to issue a new fungible ESDT token.
+	pub fn issue_fungible(
+		&self,
+		issue_cost: BigUint,
+		token_display_name: &BoxedBytes,
+		token_ticker: &BoxedBytes,
+		initial_supply: &BigUint,
+		num_decimals: usize,
+		can_freeze: bool,
+		can_wipe: bool,
+		can_pause: bool,
+		can_mint: bool,
+		can_burn: bool,
+		can_change_owner: bool,
+		can_upgrade: bool,
+		can_add_special_roles: bool,
+	) -> ContractCall<BigUint> {
+		self.issue(
+			issue_cost,
+			EsdtTokenType::Fungible,
+			token_display_name,
+			token_ticker,
+			initial_supply,
+			num_decimals,
+			can_freeze,
+			can_wipe,
+			can_pause,
+			can_mint,
+			can_burn,
+			can_change_owner,
+			can_upgrade,
+			can_add_special_roles,
+		)
+	}
+
+	/// Produces a contract call to the ESDT system SC,
+	/// which causes it to issue a new non-fungible ESDT token.
+	pub fn issue_non_fungible(
+		&self,
+		issue_cost: BigUint,
+		token_display_name: &BoxedBytes,
+		token_ticker: &BoxedBytes,
+		can_freeze: bool,
```

Comment:
We should probably group all these arguments in a ESDTMetadata struct, but it is ok for now